cg/models/downsample/downsample_data.py

```diff
@@ -0,0 +1,161 @@
+        The new sample contains the original sample internal id and meta data.
+        Raises:
+            ValueError if the downsampled sample already exists in statusDB
+        """
+        application_version: ApplicationVersion = self.get_application_version(self.original_sample)
+        downsampled_sample: Sample = self.status_db.add_sample(
+            name=self.downsampled_sample_name,
+            internal_id=self.downsampled_sample_name,
+            sex=self.original_sample.sex,
+            order=self.original_sample.order,
+            reads=multiply_by_million(self.number_of_reads),
+            downsampled_to=multiply_by_million(self.number_of_reads),
+            from_sample=self.original_sample.internal_id,
+            tumour=self.original_sample.is_tumour,
+            priority=Priority.standard,
+            customer=self.original_sample.customer,
+            application_version=application_version,
+        )
+        return downsampled_sample
+
+    def _generate_statusdb_downsampled_case(
+        self,
+    ) -> Family:
+        """Generate a case for the downsampled samples. The new case uses existing case data."""
+        if case_exists_in_statusdb(status_db=self.status_db, case_name=self.downsampled_case_name):
+            return self.status_db.get_case_by_name(self.downsampled_case_name)
+        downsampled_case: Family = self.status_db.add_case(
+            data_analysis=self.original_case.data_analysis,
+            data_delivery=self.original_case.data_delivery,
+            name=self.downsampled_case_name,
+            panels=self.original_case.panels,
+            priority=self.original_case.priority,
+            ticket=self.original_case.latest_ticket,
+        )
+        downsampled_case.customer = self.original_case.customer
+        return downsampled_case
+
+    def has_enough_reads_to_downsample(self) -> bool:
+        """Check if the sample has enough reads to downsample."""
+        if multiply_by_million(self.number_of_reads) > self.original_sample.reads:
+            raise ValueError(
+                f"Sample {self.original_sample.internal_id} does not have enough reads ({self.original_sample.reads}) to down sample to "
+                f"{self.number_of_reads}M."
+            )
+        return True
+
+    @property
+    def fastq_file_input_directory(self) -> Path:
+        """Get the latest version directory for a sample in housekeeper."""
+        LOG.debug(
+            f"Trying to get input fastq directory for {self.original_sample.internal_id} from Housekeeper."
+        )
+        return self.housekeeper_api.get_latest_bundle_version(
+            bundle_name=self.original_sample.internal_id,
+        ).full_path
+
+    @property
+    def fastq_file_output_directory(self):
+        """Get the output directory for the downsampled sample."""
+        return Path(self.out_dir, self.downsampled_sample.internal_id)
+
+    @staticmethod
```

Comment:
Remembering what we have discussed about favouring utils modules to static methods, It seems that these static functions are more suitable for a statusdb utils module, but feel free to disregard